File: app/src/main/java/dev/ragnarok/fenrir/domain/IAccountsInteractor.java
package dev.ragnarok.fenrir.domain;

import java.util.Collection;
import java.util.List;

import dev.ragnarok.fenrir.api.model.VkApiProfileInfo;
import dev.ragnarok.fenrir.api.model.response.PushSettingsResponse;
import dev.ragnarok.fenrir.model.Account;
import dev.ragnarok.fenrir.model.BannedPart;
import dev.ragnarok.fenrir.model.User;
import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Single;

public interface IAccountsInteractor {
    Single<BannedPart> getBanned(int accountId, int count, int offset);

    Completable banUsers(int accountId, Collection<User> users);

    Completable unbanUser(int accountId, int userId);

    Completable changeStatus(int accountId, String status);

    Single<Boolean> setOffline(int accountId);

    Single<VkApiProfileInfo> getProfileInfo(int accountId);

    Single<List<PushSettingsResponse.ConversationsPush.ConversationPushItem>> getPushSettings(int accountId);

    Single<Integer> saveProfileInfo(int accountId, String first_name, String last_name, String maiden_name, String screen_name, String bdate, String home_town, Integer sex);

    Single<List<Account>> getAll(boolean refresh);
}

File: app/src/main/java/dev/ragnarok/fenrir/domain/IAccountsInteractor.kt
package dev.ragnarok.fenrir.domain

import dev.ragnarok.fenrir.api.model.VkApiProfileInfo
import dev.ragnarok.fenrir.api.model.response.PushSettingsResponse.ConversationsPush.ConversationPushItem
import dev.ragnarok.fenrir.model.Account
import dev.ragnarok.fenrir.model.BannedPart
import dev.ragnarok.fenrir.model.User
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Single

interface IAccountsInteractor {
    fun getBanned(accountId: Int, count: Int, offset: Int): Single<BannedPart>
    fun banUsers(accountId: Int, users: Collection<User>): Completable
    fun unbanUser(accountId: Int, userId: Int): Completable
    fun changeStatus(accountId: Int, status: String?): Completable
    fun setOffline(accountId: Int): Single<Boolean>
    fun getProfileInfo(accountId: Int): Single<VkApiProfileInfo>
    fun getPushSettings(accountId: Int): Single<List<ConversationPushItem>>
    fun saveProfileInfo(
        accountId: Int,
        first_name: String?,
        last_name: String?,
        maiden_name: String?,
        screen_name: String?,
        bdate: String?,
        home_town: String?,
        sex: Int?
    ): Single<Int>

    fun getAll(refresh: Boolean): Single<List<Account>>
}